velox/expression/ReverseSignatureBinder.h

```diff
@@ -40,15 +40,28 @@ class ReverseSignatureBinder : private SignatureBinderBase {
   /// tryBind() and only if tryBind() returns true. If a type variable is not
   /// determined by tryBind(), it maps to a nullptr.
   const std::unordered_map<std::string, TypePtr>& bindings() const {
+    VELOX_CHECK(
+        tryBindSucceeded_, "tryBind() must be called first and succeed");
     return typeVariablesBindings_;
   }
 
+  /// Return the integer bindings produced by 'tryBind'. This function should be
+  /// called after 'tryBind' and only if 'tryBind' returns true.
+  const std::unordered_map<std::string, int>& integerBindings() const {
+    VELOX_CHECK(
+        tryBindSucceeded_, "tryBind() must be called first and succeed");
+    return integerVariablesBindings_;
+  }
+
  private:
-  /// Return whether there is a constraint on an integer variable in type
-  /// signature.
+  // Return whether there is a constraint on an integer variable in type
+  // signature.
```

Comment:
nit: do we need this change?

Reply:
I believe this is still needed.

ArgumentTypeFuzzer::fuzzArgumentTypes uses ReverseSignatureBinder to bind variables based on return type. This works well for functions like least and greatest where return type re-uses argument types as is. However, this doesn't work when return type is computed from argument types using formulas. In this case, we want ArgumentTypeFuzzer::fuzzArgumentTypes to "fail" to fuzz arguments and fallback to custom argument type generators.

Does this make sense? Do you have something else in mind?

Reply:
I did not put it clearly. I was trying to say the change from /// to // in the comment. Thanks for the explain!

Reply:
Thanks! I understand this change now.